starfish/test/test_multiprocessing.py

```diff
@@ -0,0 +1,113 @@
+"""
+These tests ensure that we can share an np-array or something that encapsulates an np-array among
+processes.  The data is shared such that any process can write to the array and it is visible to all
+the other processes.  It is up to the developer that uses this mechanism to ensure data integrity is
+maintained.
+"""
+
+
+import ctypes
+import multiprocessing
+from functools import partial
+# Even though we import multiprocessing, mypy can't find the Array class.  To avoid sprinkling
+# ignore markers all over the file, we explicitly import the symbol and put the ignore marker here.
+from multiprocessing import Array as mp_array  # type: ignore
+from typing import Any, Callable
+
+import numpy as np
+
+from starfish.multiprocessing import shmem
+
+
+def test_numpy_array(nitems: int=10):
+    """
+    Try to share a numpy array directly.  This should fail as numpy is by default a copy-on-write
+    object.  The worker process will write to array and we should not see the changes.
+    """
+    array = np.zeros(shape=(nitems,), dtype=np.uint8)
+    _start_process_to_test_shmem(array, _decode_numpy_array_to_numpy_array, nitems)
```

Comment:
suggestion: use `**kwargs` in functions defined here, this corner of python is a bit circuitous, naming things will make coming back to these tests later a bit easier

Reply:
I don't understand this suggestion.

Reply:
Sorry, I meant: 

```suggestion
    _start_process_to_test_shmem(
        array_holder=array, 
        decoder=_decode_numpy_array_to_numpy_array, 
        nitems=nitems
    )
```

Reply:
Applied.